### ISSUE TITLE
Testing jdaviz

### ISSUE THREAD

**Issue by zonca** [oirlab/iris_visualization#2]
Please follow instructions here: https://github.com/oirlab/iris_visualization#3d-visualization

Let's gather here feedback:

* was `jdaviz` easy to install?
* anything missing from my instructions?
* do you think `jdaviz` is missing functionality normally available in these tools?
* would you recommend an alternative?

**Comment by mcosens** [comment]
Setting the stretch on images in jdaviz is not as easy as it is in DS9. There's an option in DS9 to get a histogram of flux in the image and interactively set the min and max on that histogram. Something like that would be really nice to incorporate in jdaviz if possible rather than just the drop down with percentiles or a min/max input.
![Screen Shot 2021-03-22 at 1 48 21 PM](https://user-images.githubusercontent.com/22059989/112057005-f5d6c800-8b15-11eb-8890-091b734ccd0c.png)
<img width="669" alt="Screen Shot 2021-03-22 at 1 45 43 PM" src="https://user-images.githubusercontent.com/22059989/112057008-f66f5e80-8b15-11eb-9eb8-7d2cbde35b25.png">

**Comment by mcosens** [comment]
Are you able to add features to this? Overall it has some really great features but there are some improvements I'd like to see. It does show multiple fits extensions really nicely which is a great feature for looking at things like flux and uncertainties side by side, and it shows spectra a lot easier than something like ds9. There are a couple things that could be improved though like highlighting a spectral range seems to only work on the full cube spectrum rather than a highlighted spatial region. And the flux limits I posted above could be a bit more user friendly. It is a really nice base though if it can be added to or adjusted a bit.

**Comment by zonca** [comment]
definitely, it has a composable architecture. it is possible to write plugins that can add functionality without modifying the source code of `jdaviz`. However I don't have a feeling on how complex is that.
It is good to have the 2 examples you gave of improvements so I can try to implement them.

If you have other suggestions feel free to add here.

I'll get back to you if I manage to have a working prototype of one of those features.

**Comment by zonca** [comment]
## Subsets

Trying to understand better how subsets work,
so let's say I select a spatial region around a source and define a red "Subset 1":

![spatial_region](https://user-images.githubusercontent.com/383090/113614403-4adf0780-9607-11eb-9302-30d902b6a6d7.png)

Then I deselect the subset and create a green "Subset 2" in frequency:

![frequency_subset](https://user-images.githubusercontent.com/383090/113614401-47e41700-9607-11eb-9b6f-27ffeeb4064f.png)

I can switch back and forth between the 2 subsets but they are independent.

@mcosens would you like instead to make an intersection of the 2 subsets? do you have an application in mind that requires this?

**Comment by mcosens** [comment]
Yeah being able to make an intersection of the two subsets would be a nice feature to have in order to do something like fitting emission lines over a specific spatial region (maybe to get a quick line width or SNR calculation at a region of interest). For example I have data cubes with multiple HII regions in the field and I'd like to fit the [OIII] line in just one of those regions. That would be a useful tool during observing to check if anything needs to be adjusted and for preliminary analysis after.
